src/librustc_resolve/lib.rs

```diff
@@ -1665,7 +1663,7 @@ impl<'a> Resolver<'a> {
                                     let type_parameters =
                                         HasTypeParameters(&sig.generics,
                                                           FnSpace,
-                                                          MethodRibKind);
+                                                          MethodRibKind(false));
```

Comment:
I'm not sure if it's useful but in doubt, it's certainly more correct to tell wether the method is static here, rather than just using `false`. You have a `MethodSig` so it should be trivial.